encounter panic when turning on AutoKubernetesLabels

**Describe the bug**
When turning on AutoKubernetesLabels, fluent-bit log returns panic as follow



**Expected behavior**
Successfully see all logs in Loki

**Environment:**
 - Infrastructure: minikube v1.6.2
 - Deployment tool: kustomize

**Screenshots, Promtail config, or terminal output**

kubectl logs output
```
[2020/01/15 08:28:55] [ info] [filter_kube] https=1 host=kubernetes.default.svc port=443
[2020/01/15 08:28:55] [ info] [filter_kube] local POD info OK
[2020/01/15 08:28:55] [ info] [filter_kube] testing connectivity with API server...
[2020/01/15 08:28:55] [ info] [filter_kube] API server connectivity OK
[2020/01/15 08:28:55] [ info] [sp] stream processor started
panic: interface conversion: interface {} is map[string]interface {}, not map[interface {}]interface {}
```

fluent-bit.conf:

```
[SERVICE]
    Flush        1
    Daemon       Off
    Log_Level    info
    Parsers_File parsers.conf

[INPUT]
  Name tail
  Path /var/log/containers/*.log
  Parser docker
  Tag kube.*
  Refresh_Interval 5
  Mem_Buf_Limit 5MB
  Skip_Long_Lines On
  DB /run/fluent-bit/flb_kube.db

[INPUT]
  Name systemd
  Tag host.*
  Systemd_Filter _SYSTEMD_UNIT=docker.service
  Systemd_Filter _SYSTEMD_UNIT=kubelet.service
  Systemd_Filter _SYSTEMD_UNIT=node-problem-detector.service
  Max_Entries 1000
  Read_From_Tail true
  Strip_Underscores false

[FILTER]
  Name kubernetes
  Match kube.*
  Kube_Tag_Prefix kube.var.log.containers.
  Kube_URL https://kubernetes.default.svc:443
  Kube_CA_File /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
  Kube_Token_File /var/run/secrets/kubernetes.io/serviceaccount/token
  Merge_Log On
  K8S-Logging.Parser Off

[OUTPUT]
  Name loki
  Match kube.*
  Url http://loki.gohub-gohub.svc:3100/api/prom/push
  # Labels {job="fluent-bit"}
  # RemoveKeys kubernetes,stream
  AutoKubernetesLabels true
  # LabelMapPath /fluent-bit/etc/labelmap.json
  # LineFormat json
  # LogLevel warn
```

parsers.conf:

```
[PARSER]
  Name         docker
  Format       json
  Time_Key     time
  Time_Format  %Y-%m-%dT%H:%M:%S.%L
```

--- a/cmd/fluent-bit/loki.go
+++ b/cmd/fluent-bit/loki.go
@@ -2,6 +2,7 @@ package main
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
 	"os"
 	"sort"
@@ -41,7 +42,10 @@ func (l *loki) sendRecord(r map[interface{}]interface{}, ts time.Time) error {
 	level.Debug(l.logger).Log("msg", "processing records", "records", fmt.Sprintf("%+v", records))
 	lbs := model.LabelSet{}
 	if l.cfg.autoKubernetesLabels {
-		lbs = autoLabels(records)
+		err := autoLabels(records, lbs)
+		if err != nil {
+			level.Error(l.logger).Log("msg", err.Error(), "records", fmt.Sprintf("%+v", records))
+		}
 	} else if l.cfg.labelMap != nil {
 		mapLabels(records, l.cfg.labelMap, lbs)
 	} else {
@@ -65,7 +69,6 @@ func (l *loki) sendRecord(r map[interface{}]interface{}, ts time.Time) error {
 
 func toStringMap(record map[interface{}]interface{}) map[string]interface{} {
 	m := make(map[string]interface{})
-
 	for k, v := range record {
 		key, ok := k.(string)
 		if !ok {
@@ -81,36 +84,32 @@ func toStringMap(record map[interface{}]interface{}) map[string]interface{} {
 			m[key] = v
 		}
 	}
+
 	return m
 }
 
-func autoLabels(records map[string]interface{}) model.LabelSet {
-	kuberneteslbs := model.LabelSet{}
+func autoLabels(records map[string]interface{}, kuberneteslbs model.LabelSet) error {
+	kube, ok := records["kubernetes"]
+	if !ok {
+		return errors.New("kubernetes labels not found, no labels will be added")
+	}
+
 	replacer := strings.NewReplacer("/", "_", ".", "_", "-", "_")
-	for k, v := range records["kubernetes"].(map[interface{}]interface{}) {
-		switch key := k.(string); key {
+	for k, v := range kube.(map[string]interface{}) {
+		switch k {
 		case "labels":
-			for m, n := range v.(map[interface{}]interface{}) {
-				switch t := n.(type) {
-				case []byte:
-					kuberneteslbs[model.LabelName(replacer.Replace(m.(string)))] = model.LabelValue(string(t))
-				default:
-					kuberneteslbs[model.LabelName(replacer.Replace(m.(string)))] = model.LabelValue(fmt.Sprintf("%v", n))
-				}
+			for m, n := range v.(map[string]interface{}) {
+				kuberneteslbs[model.LabelName(replacer.Replace(m))] = model.LabelValue(fmt.Sprintf("%v", n))
 			}
 		case "docker_id", "pod_id", "annotations":
 			// do nothing
 			continue
 		default:
-			switch t := v.(type) {
-			case []byte:
-				kuberneteslbs[model.LabelName(k.(string))] = model.LabelValue(string(t))
-			default:
-				kuberneteslbs[model.LabelName(k.(string))] = model.LabelValue(fmt.Sprintf("%v", v))
-			}
+			kuberneteslbs[model.LabelName(k)] = model.LabelValue(fmt.Sprintf("%v", v))
 		}
 	}
-	return kuberneteslbs
+
+	return nil
 }
 
 func extractLabels(records map[string]interface{}, keys []string) model.LabelSet {

--- a/cmd/fluent-bit/loki_test.go
+++ b/cmd/fluent-bit/loki_test.go
@@ -1,6 +1,7 @@
 package main
 
 import (
+	"errors"
 	"reflect"
 	"testing"
 	"time"
@@ -295,6 +296,67 @@ func Test_labelMapping(t *testing.T) {
 			got := model.LabelSet{}
 			if mapLabels(tt.records, tt.mapping, got); !reflect.DeepEqual(got, tt.want) {
 				t.Errorf("mapLabels() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_AutoKubernetesLabels(t *testing.T) {
+	tests := []struct {
+		name    string
+		records map[interface{}]interface{}
+		want    model.LabelSet
+		err     error
+	}{
+		{
+			"records without labels",
+			map[interface{}]interface{}{
+				"kubernetes": map[interface{}]interface{}{
+					"foo": []byte("buzz"),
+				},
+			},
+			model.LabelSet{
+				"foo": "buzz",
+			},
+			nil,
+		},
+		{
+			"records with labels",
+			map[interface{}]interface{}{
+				"kubernetes": map[string]interface{}{
+					"labels": map[string]interface{}{
+						"foo":  "bar",
+						"buzz": "value",
+					},
+				},
+			},
+			model.LabelSet{
+				"foo":  "bar",
+				"buzz": "value",
+			},
+			nil,
+		},
+		{
+			"records without kubernetes labels",
+			map[interface{}]interface{}{
+				"foo":   "bar",
+				"label": "value",
+			},
+			model.LabelSet{},
+			errors.New("kubernetes labels not found, no labels will be added"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m := toStringMap(tt.records)
+			lbs := model.LabelSet{}
+			err := autoLabels(m, lbs)
+			if err != nil && err.Error() != tt.err.Error() {
+				t.Errorf("error in autolabels, error = %v", err)
+				return
+			}
+			if !reflect.DeepEqual(lbs, tt.want) {
+				t.Errorf("mapLabels() = %v, want %v", lbs, tt.want)
 			}
 		})
 	}

